include/tunnler/room_member.h

```diff
@@ -108,31 +113,45 @@ class RoomMember final {
      * Attempts to join a room at the specified address and port, using the specified nickname.
      * This may fail if the username is already taken.
      */
-    void Join(const std::string& nickname, const std::string& server = "127.0.0.1", const uint16_t serverPort = DefaultRoomPort, const uint16_t clientPort = 0);
+    void Join(const std::string& nickname, const std::string& server = "127.0.0.1",
+              const uint16_t serverPort = DefaultRoomPort, const uint16_t clientPort = 0);
 
     /**
      * Leaves the current room.
      */
     void Leave();
-    
+
 private:
-    State state;    // Current state of the RoomMember
-    MemberInformation member_information;
-    RoomInformation room_information;
+    std::atomic<State> state; ///< Current state of the RoomMember.
+    MemberList member_information; ///< Information about the clients connected to the same room as us.
```

Comment:
I think the current name is fine: "information" is non-countable. while this doesn't imply a plural from the naming, it does from the context (and so does the comment [which should probably be renamed to say ~"all room-members in the joined room" instead of client])